tss-esapi/tests/integration_tests/context_tests/tpm_commands/object_commands_tests.rs

```diff
@@ -427,3 +427,154 @@ mod test_unseal {
         assert!(unsealed == testbytes);
     }
 }
+
+mod test_create_loaded {
+    use crate::common::{create_ctx_with_session, decryption_key_pub};
+    use std::convert::{TryFrom, TryInto};
+    use tss_esapi::{
+        attributes::{ObjectAttributesBuilder, SessionAttributesBuilder},
+        constants::SessionType,
+        interface_types::{
+            algorithm::{HashingAlgorithm, PublicAlgorithm, SymmetricMode},
+            key_bits::AesKeyBits,
+            reserved_handles::Hierarchy,
+        },
+        structures::{
+            CreateLoadedKeyResult, Digest, KeyedHashScheme, PublicBuilder,
+            PublicKeyedHashParameters, SymmetricCipherParameters, SymmetricDefinition,
+            SymmetricDefinitionObject,
+        },
+    };
+
+    #[test]
+    fn test_create_loaded_tpm_alg_kdf1_sp800_108() {
+        let mut context = create_ctx_with_session();
+
+        let (session_attributes, session_attributes_mask) = SessionAttributesBuilder::new().build();
+
+        let object_attributes = ObjectAttributesBuilder::new()
+            .with_fixed_tpm(true)
+            .with_fixed_parent(true)
+            .with_st_clear(false)
+            .with_sensitive_data_origin(true)
+            .with_user_with_auth(true)
+            .with_sign_encrypt(false)
+            .with_decrypt(true)
+            .with_restricted(true)
+            .build()
+            .expect("Failed to build object attributes");
+
+        let primary_pub = PublicBuilder::new()
+            // This key is a symmetric key.
+            .with_public_algorithm(PublicAlgorithm::SymCipher)
+            .with_name_hashing_algorithm(HashingAlgorithm::Sha256)
+            .with_object_attributes(object_attributes)
+            .with_symmetric_cipher_parameters(SymmetricCipherParameters::new(
+                SymmetricDefinitionObject::AES_128_CFB,
+            ))
+            .with_symmetric_cipher_unique_identifier(Digest::default())
+            .build()
+            .unwrap();
+
+        // Create primary.
+        let primary_key_handle = context
+            .create_primary(Hierarchy::Owner, primary_pub, None, None, None, None)
+            .unwrap()
+            .key_handle;
+
+        // Create Derivation Parent
+        // - How to mark an object as a derivation parent? From what I read
+        // in the spec, a derivation parent is just when != (primary || storage)
+        let derive_parent_object_attributes = ObjectAttributesBuilder::new()
```

Comment:
It is not that simple. It also depends on what kind of handle the parent is when you call `CreateLoaded`.

Look at `TCG TPM2, r1p59, Part3 Commands, Section 12.9.1`

Reply:
Yes I did read that section. The problem I hit was that part 3 12.9.1 doesn't actually define a derivation parent - that's actually defined in part 1 4.19 as "loadable key used to derive other keys; a TPM_ALG_KEYEDHASH Parent Key" and part 1 25.1.5 table 24 states "Asymmetric keys and
symmetric keys with these attributes are Storage Parents, and hash objects with these attributes are
Derivation Parents. " where the attributes are sign=clear, decrypt=set, restricted=set. 

I think the comment I wrote here needs to be updated to reflect this, but I still think the attributes I used on the key were correct :)